tests/ticket/search.rb

```diff
@@ -108,6 +285,399 @@
+                csrf_userid: $csrf_userid,
+                csrf_token: $csrf_token,
+                page: page,
+                departments: "[1]"
+            })
+            (result['status']).should.equal('success')
+
+            tickets = result['data']['tickets']
+
+            for ticket in tickets
+                (ticket['department']['id']).should.equal('1')
+            end
+        end
+    end
+
+    it 'should success if the authors are valid' do
+        authorId1 = 1
+        authorId2 = 1
+        authorId3 = 1
+
+        authorIsStaff1 = 1
+        authorIsStaff2 = 1
+        authorIsStaff3 = 0
+
+        authorsFilter1 = [{"id": authorId1, "isStaff": authorIsStaff1}]
+        authorsFilter2 = [{"id": authorId2, "isStaff": authorIsStaff2}, {"id": authorId3, "isStaff": authorId3}, {"id": authorId1, "isStaff": authorIsStaff1}]
+        authorsFilter3 = []
+
+
+        authorId1 = authorId1.to_s
+        authorId2 = authorId2.to_s
+        authorId3 = authorId3.to_s
+
+        authorIsStaff1 = authorIsStaff1 === 1
+        authorIsStaff2 = authorIsStaff2 === 1
+        authorIsStaff3 = authorIsStaff3 === 1
+
+        authorsFilter1 = authorsFilter1.to_json
+        authorsFilter2 = authorsFilter2.to_json
+
+        for page in $pages
+            result = request('/ticket/search', {
+                csrf_userid: $csrf_userid,
+                csrf_token: $csrf_token,
+                page: page,
+                authors: authorsFilter1
+            })
+            (result['status']).should.equal('success')
+
+            tickets = result['data']['tickets']
+
+            for ticket in tickets
+                ticketAuthor = ticket['author']
+                (ticketAuthor['id']).should.equal(authorId1)
+                (ticketAuthor['staff']).should.equal(authorIsStaff1)
+            end
+
+            result = request('/ticket/search', {
+                csrf_userid: $csrf_userid,
+                csrf_token: $csrf_token,
+                page: page,
+                authors: authorsFilter1
+            })
+            (result['status']).should.equal('success')
+
+            tickets = result['data']['tickets']
+
+            for ticket in tickets
+                ticketAuthor = ticket['author']
+                ticketAuthorId = ticketAuthor['id']
+
+                if ticketAuthorId === authorId1
+                    (ticketAuthor['id']).should.equal(authorId1)
+                    (ticketAuthor['staff']).should.equal(authorIsStaff1)
+                elsif ticketAuthorId === authorId2
+                    (ticketAuthor['id']).should.equal(authorId2)
+                    (ticketAuthor['staff']).should.equal(authorIsStaff2)
+                elsif ticketAuthorId === authorId3
+                    (ticketAuthor['id']).should.equal(authorId3)
+                    (ticketAuthor['staff']).should.equal(authorIsStaff3)
+                else
+                    1.shoul.equal(0)
+                end
```

Comment:
Ditto, and in this case you have `shoul` instead of `should`, would have failed anyways I think, but this time because of bad syntax.